hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/ReflectionUtils.java

```diff
@@ -124,15 +124,29 @@ private static void setJobConf(Object theObject, Configuration conf) {
    */
   @SuppressWarnings("unchecked")
   public static <T> T newInstance(Class<T> theClass, Configuration conf) {
+    return newInstance(theClass, conf, EMPTY_ARRAY);
+  }
+
+  /** Create an object for the given class and initialize it from conf
+   *
+   * @param theClass class of which an object is created
+   * @param conf Configuration
+   * @param argTypes the types of the arguments
+   * @param values the values of the arguments
+   * @return a new object
+   */
+  @SuppressWarnings("unchecked")
+  public static <T> T newInstance(Class<T> theClass, Configuration conf,
+      Class<?>[] argTypes, Object ... values) {
```

Comment:
One more thing we can do here is perhaps checking whether `argTypes.length == values.length`, and throw `IllegalArgumentException` like: "1 parameters are required but 0 arguments are provided".

Reply:
Good idea, done.